python/ray/tune/trial_runner.py

```diff
@@ -168,20 +169,23 @@ def checkpoint(self):
         """Saves execution state to `self._metadata_checkpoint_dir`.
 
         Overwrites the current session checkpoint, which starts when self
-        is instantiated.
+        is instantiated. Only runs every 10 seconds.
         """
         if not self._metadata_checkpoint_dir:
             return
+        if time.time() - self._last_checkpoint_timestamp < 10:
```

Comment:
Should the throttle value be configurable?

Reply:
Hm, I wanted to avoid the explosion of arguments in `tune.run`, so I figured to just hard code it for now. What do you think?

Reply:
Definitely agreed that it would be nice to keep the number of `tune.run` arguments reasonable. However, I also think that it would really suck if someone needed to get checkpoints, say, every 5 seconds and they just couldn't do that. For us this wouldn't be a problem since we checkpoint so infrequently.

Reply:
Another way to do this could maybe be warning the user and telling them to decrease the checkpoint frequency?

Reply:
Oh, this is experiment checkpoints for experiment fault tolerance, not individual trial checkpoints. Presumably you're talking about users needing to access their trial checkpoints every 5 seconds? IIUC, I think this shouldn't affect that since the trial checkpoint/rsyncing mechanism is done separately (but not sure if that's what you mean)

Right now, we write the a metadata checkpoint of the entire experiment to disk every iteration of every trial, which becomes a huge bottleneck if running like 1000 trials, each with pretty short iteration times. 

One option for user to control this is to have an env var check. 

What do you think?

Reply:
Oh I see. You're right, I got this mixed with the individual trial checkpointing. For experiment level checkpointing, I think this makes much more sense, and I think it's fine even if this could not be configured.